components/pedigree/ui/src/main/resources/PhenoTips/PedigreeImageService.xml

```diff
@@ -37,13 +37,36 @@
   <minorEdit>false</minorEdit>
   <syntaxId>xwiki/2.1</syntaxId>
   <hidden>true</hidden>
-  <content>{{include reference="PhenoTips.PedigreeMacros" /}}
-
+  <content>
 {{velocity}}
+#if ("$!request.action" == 'export')
+  {{html wiki="false" clean="false"}}
+  $response.setContentType('application/json')##
+  #if ("$!request.image" == '')
+    $response.setStatus(404)##
+    {"error" : "$services.localization.render('phenotips.PedigreeImageService.noPedigreeImage')"}
+    #break
+  #end
+  ## Fix broken SVG generated by a broken version of the pedigree editor
+  #if ($stringtool.countMatches($request.image, 'xmlns=') &gt; 1)
```

Comment:
There is a reason, indeed. Putting the fix only in the JavaScript runtime means that a pedigree that's already broken has to be re-opened and re-saved to get it fixed, while this is just a workaround so that the users have a more pleasant experience. But I agree that the proper fix has to be in the JS code, so that broken SVGs aren't generated at all.